vault/plugin_catalog.go

```diff
@@ -104,6 +97,28 @@ func (c *PluginCatalog) getPluginTypeFromUnknown(ctx context.Context, logger log
 	return consts.PluginTypeUnknown, nil
 }
 
+func isDatabasePlugin(ctx context.Context, plugin *pluginutil.PluginRunner, logger log.Logger) error {
+	merr := &multierror.Error{}
+	// Attempt to run as database V5 plugin
+	v5Client, err := newdbplugin.NewPluginClient(ctx, nil, plugin, log.NewNullLogger(), true)
+	if err == nil {
+		// Close the client and cleanup the plugin process
+		v5Client.Close()
+		return nil
+	}
+	merr = multierror.Append(merr, err)
```

Comment:
Does the error from there indicate which database version was attempted? If not, might be useful to wrap the error message with this bit of info.

Reply:
It does not but that's because the error will only be returned if both v5 and v4 fail.

Holding off on doing anything with this until we figure out what to do with the log message that this error is used in.

Reply:
Wrapped the error to indicate which error failed on which database version. This will bubble up to the log message referenced in the next comment thread.